drivers/platform/x86/hp-wmi.c

```diff
@@ -275,9 +287,13 @@ static int hp_wmi_als_state(void)
 static int hp_wmi_dock_state(void)
 {
 	int state = 0;
-	int ret = hp_wmi_perform_query(HPWMI_HARDWARE_QUERY, 0, &state,
-				       sizeof(state), sizeof(state));
+	int ret;
+
+	if (dmi_check_system(not_dockable))
+		return 0;
```

Comment:
I'm not sure of how much a dmi call cost, but I believe it involves iterating a list, right? wouldn't be better to have this in the module init/probe, set a flag, and just check the flag here?